test/memory/memory_api_test.go

```diff
@@ -0,0 +1,191 @@
+//  Copyright (c) 2024 Cisco and/or its affiliates.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at:
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package memory
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"go.fd.io/govpp/binapi/vpe"
+	"go.fd.io/govpp/test/vpptesting"
+	"runtime"
+	"runtime/metrics"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+)
+
+var apiNum = flag.Uint("api-num", 0, "Custom number API to test")
+
+// required go runtime metrics
+const (
+	heapBytesAllocs  = "/gc/heap/allocs:bytes"
+	heapBytesFrees   = "/gc/heap/frees:bytes"
+	heapObjectAllocs = "/gc/heap/allocs:objects"
+	heapObjectFrees  = "/gc/heap/frees:objects"
+	goroutines       = "/sched/goroutines:goroutines"
+)
+
+// TestAPIMemory tests the GoVPP memory consumption for various numbers of API calls
+func BenchmarkAPIMemory(b *testing.B) {
+	flag.Parse()
+
+	fmt.Printf("Running GoVPP API calls memory test\n\n")
+	test := vpptesting.SetupVPP(b)
+	vpeRPC := vpe.NewServiceClient(test.Conn)
+
+	samples := []metrics.Sample{
+		{Name: heapBytesAllocs},
+		{Name: heapBytesFrees},
+		{Name: heapObjectAllocs},
+		{Name: heapObjectFrees},
+		{Name: goroutines},
+	}
+	pass := true
+	testAPICalls := func(n uint, thresholds [3]uint64) {
+		tHolds0, tHolds1, tHolds2 := "N/A", "N/A", "N/A"
+
+		now := time.Now()
+
+		fmt.Printf("For %d:\tMeasured\tThreshold\n%s\n", n, strings.Repeat("-", 41))
+
+		runtime.GC()
+		metricsBefore := readMetrics(samples)
+		for i := 0; i < int(n); i++ {
+			if _, err := vpeRPC.ShowVersion(context.Background(), &vpe.ShowVersion{}); err != nil {
+				b.Fatal("calling show version failed:", err)
+			}
```

Comment:
Agreed, although selecting the right calls messages might be a bit tricky...

Reply:
Isn't every API using use the same underlying code? I chose the `ShowVersion` because it is fast, and can be called in a loop indefinitely. I can add some other scenarios (like loop interfaces, and switch between add/delete so it won't overflow anything). 
If you have some specific APIs in mind, let me know.